### PR TITLE
Add more tests for image decoding/encoding part

### PR DESCRIPTION
**Pull request by dreampiggy**
### New Pull Request Checklist

* [x] I have read and understood the [CONTRIBUTING guide](https://github.com/rs/SDWebImage/blob/master/.github/CONTRIBUTING.md)
* [x] I have read the [Documentation](http://cocoadocs.org/docsets/SDWebImage/)
* [x] I have searched for a similar pull request in the [project](https://github.com/rs/SDWebImage/pulls) and found none

* [x] I have updated this branch with the latest master to avoid conflicts (via merge from master or rebase)
* [x] I have added the required tests to prove the fix/feature I am adding
* [x] I have updated the documentation (if necessary)
* [x] I have run the tests and they pass
* [x] I have run the lint and it passes (`pod lib lint`)

This merge request fixes / reffers to the following issues: ...

### Pull Request Description

This PR does not introduce any code for framework, but only for test case. Our code coverage is drop during 4.x -> 5.x, I think it's time to catch up, the aim is at least 80%.

This PR contains some test case about image decoding/encoding.

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -1,6 +1,6 @@
 
 language: objective-c
-osx_image: xcode9
+osx_image: xcode9.4
```
Comment (bpoplauschi) [inline .travis.yml]: it was the time to update the osx_image and use a newer simulator

**Comment by dreampiggy** [comment]
Maybe I get the reason. It seems that Travis-CI use some low hard-ware machine, which does not supports HEIC encoding (See the requirement: ). However, on my local MacBook it works :)

I can temporarily disable encoding check on macOS.

**Comment by bpoplauschi** [comment]
Can't we just add `if canDecodeFromHEICFormat` to the test?

**Comment by dreampiggy** [comment]
Emmm...Yes we can. This `test13ThatHEICWorks` actually contribute to code coverage, are 3 methods: `sd_UTTypeFromImageFormat` && `canDecodeFromHEICFormat` && `canEncodeToHEICFormat`. The other are also covered by JPEG or PNG because we use Image/IO for encoding/decoding, and it's transparent for us (unless Apples contains bug...)

But I think it's also a check, to avoid us use some strange API that cause HEIC decoding/encoding faild for some user.

**Comment by dreampiggy** [comment]
Since this is a test PR, it can increase the code coverage to 5%. I'll merge it because it does not introduce anything on the framework side.

One little issue is that I temporarily disable the `HEIF` encoding compatibility check on Travis-CI for macOS, because of its own issue. However, if future OSX image is available, we can also upgrade and re-enable it.

**Comment by bpoplauschi** [comment]
Thanks @dreampiggy

**Pull request merged**